### PR TITLE
DEVEXP-487: E2E Conversation/TemplatesV2

### PR DESCRIPTION
**Pull request by asein-sinch**

### CODE REVIEW

File: packages/conversation/tests/rest/v1/templates-v2/templates-v2.steps.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  assert.equal(templatesList.templates?.length, 2);
+  for(const templateV2 of templatesList.templates!) {
+    assert.ok(templateV2.version);
+    let latestVersionCount = 0;
+    let otherVersionCount = 0;
+    const translations = templateV2.translations!;
+    for(const translation of translations) {
+      translation.version === 'latest' ? latestVersionCount++ : otherVersionCount++;
+    }
+    assert.equal(latestVersionCount, otherVersionCount);
```
Comment (JPPortier) [inline packages/conversation/tests/rest/v1/templates-v2/templates-v2.steps.ts:70]: Not having yet managed templates, I'm onboarding about this test validation and discovering by reading the test node source itself: it is performing some business logic validation not described by the test content description: `the response contains the list of conversation templates with the V2 structure`
So when I'll write the Java e2e, I could be not fully aware about this test content validation.
Can we imagine having a test description ensuring we will validate properly this business logic ?

File: packages/conversation/src/models/v1/requests/templates-v2/templates-v2-request-data.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { V2Template } from '../../v2-template';
 
 export interface V2CreateTemplateRequestData {
   /** Required. The template to create. */
-  'createTemplateRequestBody': V2Template;
+  'createTemplateRequestBody': Omit<V2Template, 'create_time' | 'update_time'>;
```
[the remark this answers is not in the document]
Comment (asein-sinch) [reply packages/conversation/src/models/v1/requests/templates-v2/templates-v2-request-data.ts:5]: > as a Java developer not having the 'Omit' keyword

Life is unfair 😆

[the remark this answers is not in the document]
Comment (asein-sinch) [reply packages/conversation/src/models/v1/requests/templates-v2/templates-v2-request-data.ts:5]: I beg to differ: this approach would need to calculate the GCD for each schema update, resulting in a base schema that has little meaning by itself. In your reasoning you take the assumption that a new field will necessarily be common to all schema, which is not guaranteed.
With this approach, I try to put the REST object back into the center of the API to improve the developer experience: a user will have a plain view of the object properties by just looking at the `V2Template` interface.
Then, when creating or modifying such object, they will be deprived of the read-only field:
 - create_time and update_time when creating a Template (the id can be set manually as a ULID)
 - id, create_time and update_time when updating a Template
 
 Let's take 2 examples:

1. **A new read-only field `trace_id` is added**
    - Java-like approach: it cannot be added in the base class, but only in the schema(s) describing a response object
    - TS-like approach: the property will be added to the V2Template interface and then added to the `Omit` list for all the modifying operations

1. **A new field `author` is added**
    - Java-like approach: as the field can be read and edited, it goes to the base class
    - TS-like approach: the property will be added to the V2Template interface as it aims at describing all the properties of the object

Conclusion: The impact on the developer side (that I'm willing to take) will be equivalent in both ways: maintenance-wise, these 2 approaches only differ when the API will add a new read-only field (i.e. trace_id): the SDK developer will have to scan all the classes impacted and in the Java-way update the response objects, contrary to the TS-way where he will have to update the `Omit` list of the request objects.

File: packages/conversation/tests/rest/v1/templates-v2/templates-v2.steps.ts (hunk 1 above)
Comment (asein-sinch) [reply packages/conversation/tests/rest/v1/templates-v2/templates-v2.steps.ts:70]: Right, it deserves its own statement

Comment (asein-sinch) [reply packages/conversation/tests/rest/v1/templates-v2/templates-v2.steps.ts:74]: Fixed

File: packages/conversation/src/models/v1/requests/templates-v2/templates-v2-request-data.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JPPortier) [reply packages/conversation/src/models/v1/requests/templates-v2/templates-v2-request-data.ts:5]: > you take the assumption that a new field will necessarily be common to all schema, which is not guaranteed.

Not exactly, "my" logic/explanation if about a field not common to all schema.
And was not a blocker for PR.
About GCD: yes it exactly the purpose.
Research to provide it is performed could be provided inheritance (so an "adding" fields) for Java; and is provided by "Omit" (so a "substraction" of fields) for TS.
In both cases: SDK's writer is performing a GDC research to provide better DX

> as a Java developer not having the 'Omit' keyword

So, we can now saying we had proof that Java's developer is more positive than TS one 😄 
 
> With this approach, I try to put the REST object back into the center of the API to improve the developer experience: a user will have a plain view of the object properties by just looking at the `V2Template` interface. 

Just to troll (and again initial was not a blocker for this PR, just to share some thoughts; we known enough each other scope of aim to provide DX)
 One point for Java SDK providing online javadoc: this is where sum of all inheritance will take place for SDK's user having overview of "V2Template" class

**Review by JPPortier — APPROVED** [review]

**Pull request merged**